### DISCUSSION TITLE
Use gofumpt as Go formatter

### DISCUSSION THREAD

**Discussion by ilius** [zed-industries/zed#20779] [Show and tell]
Here is how to use [gofumpt](https://github.com/mvdan/gofumpt) as Go formatter:
```json
	"languages": {
		"Go": {
			"format_on_save": "on",
			"formatter": {
				"external": {
					"command": "gofumpt",
					"arguments": ["{buffer_path}"]
				}
			}
		}
	}

```

**Comment by praktiskt** [comment]
Hi, I wasn't able to get `gofumpt` running in Zed like this. There seems to be a conflict if you use `gopls` too. 

I found [this post by Niklas Metje](https://niklasmtj.de/blog/use-gofumpt-formatter-in-zed-as-golang-default/) explaining an alternative way to configure `gofumpt` in Zed using `gopls`:

```json
  "lsp": {
    "gopls": {
      "gofumpt": true,
      "initialization_options": {
        "gofumpt": true
      }
    }
  }
```

This worked perfectly for me.

**Reply by fgimian** [reply]
Actually appears that you only need it under `initialization_options` based on my own testing:

i.e.

```json
{
  "lsp": {
    "gopls": {
      "initialization_options": {
        "gofumpt": true
      }
    }
  },
  ...
}
```
